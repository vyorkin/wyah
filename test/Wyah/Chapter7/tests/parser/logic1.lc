-- Logic 1

let not p = p F T;
let and p q = p q F;
let or p q = p T q;
let cond p x y = p x y;
