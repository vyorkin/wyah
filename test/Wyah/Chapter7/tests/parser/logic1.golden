Program [Decl "cond" (ELam (Var "y") (ELam (Var "x") (ELam (Var "p") (EApp (EApp (EVar (Var "p")) (EVar (Var "x"))) (EVar (Var "y")))))),Decl "or" (ELam (Var "q") (ELam (Var "p") (EApp (EApp (EVar (Var "p")) (EVar (Var "T"))) (EVar (Var "q"))))),Decl "and" (ELam (Var "q") (ELam (Var "p") (EApp (EApp (EVar (Var "p")) (EVar (Var "q"))) (EVar (Var "F"))))),Decl "not" (ELam (Var "p") (EApp (EApp (EVar (Var "p")) (EVar (Var "F"))) (EVar (Var "T"))))]